lightning/src/ln/onchaintx.rs

```diff
@@ -709,4 +778,47 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 			self.pending_claim_requests.remove(&req);
 		}
 	}
+
+	pub(super) fn provide_latest_local_tx(&mut self, tx: LocalCommitmentTransaction) -> Result<(), ()> {
```

Comment:
Using a result here is the lazy option - this should be infallible and we should panic on programming errors which are fundamentally unsafe. At a high level, I think the right approach is to

 * gate ChannelMonitor requesting such transactions on the channel having been closed,
 * have ChannelManager close the channel on would_broadcast_at_height(height + 2),
 * then have ChannelMonitor close it at the regular height (allowing itself to "close" the channel).

That way we shouldn't have any races where ChannelMonitor tries to get the latest local transaction but the channel isn't closed.

Then,
 * ChannelMonitor::get_latest_local_commitment_transaction should return a Result (since its the public API),
 * If a channel is already force-closed, we should allow it to return fine,
 * If a channel is still open - we should return an Err and the client should force-close the channel in the manager (or require a ChannelMonitor::channel_was_force_closed() call).